app/controllers/tree_controller.rb

```diff
@@ -0,0 +1,18 @@
+# This is a highly experimental implementation of something that we would like to have probably in an UI-API
+# It is definitely not a good example and it SHOULD NOT BE COPY-PASTED in any case
+class TreeController < ApplicationController
+  before_action :check_privileges
+
+  def automate_entrypoint
+    tree = TreeBuilderAutomateEntrypoint.new(:automate_entrypoint, :automate_entrypoint_tree, {})
+    if params[:id]
+      json = TreeBuilder.convert_bs_tree(tree.x_get_child_nodes(params[:id]))
+    else
+      # Unfortunately the TreeBuilder does not provide a non-JSON-ized version so it needs to be reparsed
+      # TODO: Move the JSON-ization further in the TreeBuilders so this re-parsing can be removed
+      json = JSON.parse(tree.instance_variable_get(:@bs_tree))
+    end
+
+    render :json => json
```

Comment:
you can set the response headers (content-type etc) and send the json as :text instead of repartsing